<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>Shaking of an Android device is possible in all circumstances with a call to <code>getSystemService(Context.VIBRATOR_SERVICE)</code> (API 26).
    Behind this effect stands a specific miniature hardware component, motor or actuator, that consumes power. As a consequence,
    its usage must be discouraged, especially since its added value is not clear.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    Vibrator v = (Vibrator) getSystemService(Context.VIBRATOR_SERVICE);
    long[] pattern = {0, 100, 1000};
    v.vibrate(pattern, 0);
</pre>
<h2>Compliant Code Example</h2>
<pre>
    //
</pre>