<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>Giving informations to the end-user through notifications is an important aspect of a modern app.<br>
	However, a notification does not necessarily need to be loud and vibrant to achieve its purpose.<br>
	Default mode is enough. That is why when building a notification with
	<code>NotificationManager</code><br>
	there should be no extra calls to the methods <code>setVibrationPattern()</code> nor
	<code>setSound()</code> on the builder object.</p>
<h2>Noncompliant Code Example</h2>
<pre>
	context = new Context context;
	NotifManager = (NotificationManager) context;
	NotifManager.setVibrationPattern(new long[]{1000, 1000, 1000, 1000, 1000});
	NotifManager.setSound(RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION), getAudioAttributes());
</pre>