<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>Shaking of an Android device is possible in all circumstances with a call to <code>getSystemService(Context.VIBRATOR_SERVICE)</code> (API 26).
    Behind this effect stands a specific miniature hardware component, motor or actuator, that consumes power. As a consequence,
    its usage must be discouraged, especially since its added value is not clear.</p>
<h2>Noncompliant Code Example</h2>
<pre>
API 26:
  Vibrator v = (Vibrator) getSystemService(Context.VIBRATOR_SERVICE);
  v.vibrate(400);
API 31:
  VibratorManager vm = (VibratorManager) getSystemService(Context.VIBRATOR_MANAGER_SERVICE)
  Vibrator v = (Vibrator) vm.getDefaultVibrator();
  v.vibrate(400);
</pre>
<h2>Compliant Code Example</h2>
<pre>
    //
</pre>